src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -227,6 +225,50 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
             }
         }
 
+        private void CacheMember(
+            Type declaringType,
+            Type memberType,
+            MemberInfo memberInfo,
+            Dictionary<string, JsonPropertyInfo> cache,
+            ref Dictionary<string, MemberInfo>? ignoredMembers)
```

Comment:
Why is this passed by ref?

Reply:
Because it's `null` by default and assigned only in this method on line 268.